pkg/tfgen/installation_docs.go

```diff
@@ -477,3 +503,39 @@ func getProviderDisplayName(g *Generator) string {
 	capitalize := cases.Title(language.English)
 	return capitalize.String(providerName)
 }
+
+func removeEmptyExamples(contentStr string) (string, error) {
+	if checkExamplesEmpty(contentStr) {
+		mybytes, err := SkipSectionByHeaderContent([]byte(contentStr), func(headerText string) bool {
+			return headerText == "Example Usage"
+		})
+		contentStr = string(mybytes)
+		return contentStr, err
+	}
+	return contentStr, nil
+
+}
```

Comment:
3 nits:
1. Let's operate in `[]byte` space (instead of `string`) space as much as possible, since that's what `goldmark` uses.
2. Postfixing a variable name by its type should be avoided if possible.
3. Let's favor returning early if possible.
```suggestion
func removeEmptyExamplesSection(content []byte) ([]byte, error) {
	const title = "Example Usage"
	if !isMdSectionEmpty(title, content) {
		return content
	}
	isExamplesSection := func(s string) bool { return title == s }
	return SkipSectionByHeaderContent(content, isExamplesSection)
}
```